crypto/asn1/d2i_pu.c

```diff
@@ -61,6 +71,11 @@ EVP_PKEY *d2i_PublicKey(int type, EVP_PKEY **a, const unsigned char **pp,
 #endif
 #ifndef OPENSSL_NO_EC
     case EVP_PKEY_EC:
+        if (copy != NULL) {
+            /* use downgraded parameters from copy */
+            ret->pkey.ec = copy->pkey.ec;
```

Comment:
What happens with the previous `ret->pkey.ec`, if there is one?

Reply:
There is none, it is cleared up by the forced EVP_PKEY_set_type() call if copy != NULL.